src/AutorestSwift/View Models/RequestViewModel.swift

```diff
@@ -50,16 +49,24 @@ struct RequestViewModel {
         self.mediaTypes = httpWithBodyRequest?.mediaTypes ?? []
         self.knownMediaType = httpWithBodyRequest?.knownMediaType.rawValue ?? ""
 
-        // check if the request body is from signature parameter. If yes, store the object type
-        // and add the request signature parameter to operation parameters
-        var params = [ParameterViewModel]()
-        for param in request.signatureParameters ?? [] {
-            params.append(ParameterViewModel(from: param))
+        // TODO: only support the first signature parameter in Reqest now
+        let firstSignatureParam = request.signatureParameters?.first
+        self.objectType = firstSignatureParam?.schema.name
+        if isRequired(param: firstSignatureParam) {
+            self.objectName = firstSignatureParam?.name
+        } else {
+            self.objectName = "options?.\(firstSignatureParam?.name ?? "")"
         }
```

Comment:
I guess I still don't understand this logic. What scenarios is it trying to describe?

Reply:
I didn't see your earlier reply, so I do understand this a little better now. I think we should clarify in the comment that we are looking for a body-encoded parameter, and then we should filter the signatureParams by whether they have `http.in == .body`. Since we only expect one, we should assert that there's only one and then continue with our assumption. This ensures the code will crash if the assumption is invalidated but only when i development mode.